### PR TITLE
Kustomize argocd

### PR DESCRIPTION
**Pull request by HumairAK**
See commit messages for more details.

### CODE REVIEW

File: manifests/overlays/dev/configs/argo_cm/dex.config [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    id: openshift
+    name: OpenShift
+    config:
+      issuer: API_SERVER
+      clientID: system:serviceaccount:aicoe-argocd-dev:argocd-dex-server
+      clientSecret: $dex.serviceaccount.clientSecret
+      redirectURI: ARGOCD_ROUTE/api/dex/callback
+      insecureCA: true
+      groups:
+        - dev-group
```
Comment (anishasthana) [inline manifests/overlays/dev/configs/argo_cm/dex.config:13]: Missing newline

File: manifests/overlays/dev/configs/argo_cm/kustomize.buildOptions [hunk 2]
```diff
@@ -0,0 +1 @@
+--enable_alpha_plugins
```
Comment (anishasthana) [inline manifests/overlays/dev/configs/argo_cm/kustomize.buildOptions:1]: Missing newline

File: manifests/overlays/dev/configs/argo_cm/url [hunk 3]
```diff
@@ -0,0 +1 @@
+ARGOCD_ROUTE
```
Comment (anishasthana) [inline manifests/overlays/dev/configs/argo_cm/url:1]: Missing newline

File: manifests/overlays/dev/configs/argo_rbac_cm/policy.csv [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+p, role:admin, repositories, delete, *, allow
+p, role:admin, projects, create, *, allow
+p, role:admin, projects, update, *, allow
+p, role:admin, projects, delete, *, allow
+p, role:admin, accounts, update, *, allow
+p, role:admin, gpgkeys, create, *, allow
+p, role:admin, gpgkeys, delete, *, allow
+
+g, role:admin, role:readonly
+g, dev-group, role:admin
```
Comment (anishasthana) [inline manifests/overlays/dev/configs/argo_rbac_cm/policy.csv:37]: Missing newline

File: manifests/overlays/prod/configs/argo_cm/dex.config [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    config:
+      issuer: https://api.ocp.prod.psi.redhat.com:6443
+      clientID: system:serviceaccount:aicoe-argocd:argocd-dex-server
+      clientSecret: $dex.serviceaccount.clientSecret
+      redirectURI: https://argocd-server-aicoe-argocd.apps.ocp.prod.psi.redhat.com/api/dex/callback
+      insecureCA: true
+      groups:
+        - data-hub-openshift-admins
+        - aicoe-thoth-devops
+        - aicoe-aiops-devops
```
Comment (anishasthana) [inline manifests/overlays/prod/configs/argo_cm/dex.config:15]: Missing newline

File: manifests/overlays/prod/configs/argo_cm/kustomize.buildOptions [hunk 6]
```diff
@@ -0,0 +1 @@
+--enable_alpha_plugins
```
Comment (anishasthana) [inline manifests/overlays/prod/configs/argo_cm/kustomize.buildOptions:1]: Missing newline

File: manifests/overlays/prod/configs/argo_cm/resource.exclusions [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  - "*"
+- apiGroups:
+  - "*"
+  kinds:
+  - CustomResourceDefinition
+  - ClusterRole
+  clusters:
+  - https://api.ocp.prod.psi.redhat.com:6443
+  - https://datahub.psi.redhat.com:443
+  - https://paas.stage.psi.redhat.com:443
```
Comment (anishasthana) [inline manifests/overlays/prod/configs/argo_cm/resource.exclusions:16]: Missing newline

File: manifests/overlays/prod/configs/argo_cm/resource.inclusions [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+  clusters:
+  - https://paas.stage.psi.redhat.com:443
+- apiGroups:
+  - argoproj.io
+  kinds:
+  - EventSource
+  - Gateway
+  - Sensor
+  clusters:
+  - https://paas.stage.psi.redhat.com:443
```
Comment (anishasthana) [inline manifests/overlays/prod/configs/argo_cm/resource.inclusions:323]: Missing newline

File: manifests/overlays/prod/configs/argo_cm/url [hunk 9]
```diff
@@ -0,0 +1 @@
+https://argocd-server-aicoe-argocd.apps.ocp.prod.psi.redhat.com
```
Comment (anishasthana) [inline manifests/overlays/prod/configs/argo_cm/url:1]: Missing newline

File: manifests/overlays/prod/configs/argo_rbac_cm/policy.csv [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+g, data-hub-openshift-admins, role:data-hub-admin
+
+p, role:aiops-admin, applications, get, aiops/*, allow
+p, role:aiops-admin, applications, update, aiops/*, allow
+p, role:aiops-admin, applications, delete, aiops/*, allow
+p, role:aiops-admin, applications, sync, aiops/*, allow
+p, role:aiops-admin, applications, override, aiops/*, allow
+p, role:aiops-admin, applications, action/*, aiops/*, allow
+g, aicoe-aiops-devops, role:argocd-admin
+g, aicoe-aiops-devops, role:aiops-admin
```
Comment (anishasthana) [inline manifests/overlays/prod/configs/argo_rbac_cm/policy.csv:49]: Missing newline

**Comment by HumairAK** [comment]
The lack of newlines in the configs directory are intentional.

**Pull request merged**